blog/2025/pcsx2-2.6/index.mdx

```diff
@@ -0,0 +1,606 @@
+
+#### Why this is faster?
+
+- Example #1: In the first case batching means instead of issuing 5 copies/barriers for 5 triangles separately we can batch all 5 if they don't overlap each other and issue only 1 copy/barrier which is much faster.
+
+- Example #2: In the second case where we only need to copy the region we are effectively reducing the copy area that we need to copy, the smaller the area we need to copy the faster it is.
+
+<Image src={require("./img/benches/13354-Gradius-pt1.webp").default} />
+
+<Image src={require("./img/benches/13354-Gradius-pt2.webp").default} />
+
+<Image src={require("./img/benches/13354-CODFH-pt3.webp").default} />
+
+Next up in [#13354](https://github.com/PCSX2/pcsx2/pull/13354), TJnotJT also implemented another improvement which detects triangles-are-quads in primitive overlap which further reduces copies/barriers on all renderers:
+
+1. Added utility functions for detecting when vertices form axis-aligned right triangles or quads.
+2. Updated primitive overlap detection to account for pairs of triangles that together form axis-aligned quads.
+
+<Image src={require("./img/benches/13305-Jak-pt1.webp").default} />
+
+<Image src={require("./img/benches/13305-Jak-pt2.webp").default} />
+
+<Image src={require("./img/benches/13305-SH-pt3.webp").default} />
+
+#### But wait there's more!
+
+A while back, we added extra barriers to fix some bugs in Time Crisis. In [#13595](https://github.com/PCSX2/pcsx2/pull/13595) TellowKrinkle discovered that those barriers were getting added on all GPU vendors. Upon further inspection, it appears as an NVIDIA driver bug, rather than actual issues with our OpenGL API usage. This allowed us to cut down the barriers on OpenGL on other GPU vendors by a lot.
+
+Here are the top 5 highest barrier reductions we have found:
+
+```ini
+WRC_4_SCES-52389_20230329024724
+Barriers: -6604 [13473=>6869]
+
+nfsmw_reflections
+Barriers: -2125 [4250=>2125]
+
+007 Agent Under Fire - door blend
+Barriers: -1646 [3284=>1638]
+
+Ace_Combat_04_-_Shattered_Skies_SLUS-20152_20230214003004
+Barriers: -1304 [1306=>2]
+
+Avatar_-_The_Last_Airbender_SLUS-21395_20240428221833
+Barriers: -999 [17716=>16717]
+```
+
+:::info
+The less barriers there is, the better!
+:::
+
+#### More, more!
+
+Following the adjustments by TellowKrinkle, LightningTerror has discovered that we can further cut down the barriers on OpenGL in [#13576](https://github.com/PCSX2/pcsx2/pull/13576) to only issue barriers for framebuffer optimizations if needed. To check if the state changed previous draw/pass and if not, only then do we need to issue a barrier or render pass to ensure writes.
+
+We basically have an optimization where we replace render passes with barriers in some cases to ensure writes happen, but there were cases where we issued both at the same time, so in this case if we issued a render pass state change then we don't really need barriers.
```

Comment:
I don't think this is required. We went into details about the optimization above, and now we have another line doing the same thing. Or is it a different optimization? If so, it should be a list.

Here is my suggestion anyways
```suggestion
The optimization involved the case where we were issuing both a render pass and barrier. The render pass already behaves as a barrier, so we skip the barrier and only do a render pass.
```